pkg/platform/model/buildplanner.go

```diff
@@ -385,6 +395,42 @@ func (bp *BuildPlanner) CreateProject(params *CreateProjectParams) (strfmt.UUID,
 	return resp.ProjectCreated.Commit.CommitID, nil
 }
 
+type MergeCommitParams struct {
+	Owner     string
+	Project   string
+	TargetRef string // the commit ID or branch name to merge into
+	OtherRef  string // the commit ID or branch name to merge from
+	Strategy  model.MergeStrategy
+}
+
+func (bp *BuildPlanner) MergeCommit(params *MergeCommitParams) (strfmt.UUID, error) {
```

Comment:
I can see this endpoint being used for more than just FastForward strategies, so the returned commitID may be useful.